presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoResultSet.java

```diff
@@ -1816,6 +1819,37 @@ private ResultsPageIterator(StatementClient client, Consumer<QueryStats> progres
         }
     }
 
+    private static class LengthLimitedIterator<T>
```

Comment:
Make this class `final` and move it to be a top-level package-private class (since `PrestoResultSet` is already quite long)